Demo/src/main/java/com/paypal/android/viewmodels/PayPalNativeViewModel.kt

```diff
@@ -75,7 +92,43 @@ class PayPalNativeViewModel @Inject constructor(
             shippingChangeData: ShippingChangeData,
             shippingChangeActions: ShippingChangeActions
         ) {
-            // implement
+            val options: List<Options>
+            val updatedShippingAmount: String?
+
+            when (shippingChangeData.shippingChangeType) {
+                ShippingChangeType.OPTION_CHANGE -> {
+
+                    options = shippingChangeData.shippingOptions
+                    updatedShippingAmount = shippingChangeData.selectedShippingOption?.amount?.value
+                }
+                ShippingChangeType.ADDRESS_CHANGE -> {
+                    options = shippingChangeData.shippingOptions.map {
+                        it.copy(
+                            amount = it.amount?.copy(
+                                value = ((it.amount?.value?.toFloat() ?: 0f) + SHIPPING_METHOD_INCREASE).asValueString()
+                            )
+                        )
+                    }
+                    updatedShippingAmount = options.find { it.selected }?.amount?.value
+                }
+            }
+
+            val patchRequest = PatchOrderRequest(
+                PatchShippingOptions.Replace(
+                    purchaseUnitReferenceId = "PUHF",
+                    options = options
+                ),
+                PatchAmount.Replace(
+                    purchaseUnitReferenceId = "PUHF",
+                    amount = getAmount(
+                        value = order.purchaseUnitList.first().amount.value,
+                        shippingValue = updatedShippingAmount ?: "0.00"
+                    )
+                )
+            )
+            shippingChangeActions.patchOrder(patchRequest) {
```

Comment:
What does it mean for a merchant to patch an order? Basically to amend an existing order?

Reply:
I think so. Its for when a shipping address is changed or a shipping method. The order needs to be updated with the new value.